src/datapump/jobs/version_update.py

```diff
@@ -173,6 +190,42 @@ def _create_aux_tile_set(self, tile_set_parameters: RasterTileSetParameters) ->
 
         return data["asset_id"]
 
+    def _create_cog_asset(self, cog_asset_parameters: CogAssetParameters) -> str:
+        """
+        Create cog asset and return asset ID, empty string if an error
+        """
+        client = DataApiClient()
+
+        co = cog_asset_parameters
+
+        assets = client.get_assets(self.dataset, self.version)
+        asset_id = ""
+        for asset in assets:
+            if asset["asset_type"] == "Raster tile set" and f"/{co.source_pixel_meaning}/" in asset["asset_uri"]:
```

Comment:
Why don't you just check the pixel meaning on the asset instead of looking for it in the URI? I think it should be on there, or you might need to get the creation options. I think might be a little neater and less prone to error if two pixel meanings have overlapping names (e.g. `date` and `date_conf`)

Reply:
Thanks for the review!  I was just trying to avoid one extra Data API access.  Added a new simple function to access the `creation_options` of an asset, and change the test to compare directly with the `pixel_meaning` of the asset.